### PR TITLE
[WIP] Remote control

### PR DESCRIPTION
**Pull request by gertjaap**
This implements #235 . The idea is as follows:

1. You generate a EC Keypair
2. You then authorize the public key in the lit node using the `rcauth {pubkey} true` command
3. You can then sign a JSON request with your key. JSON is formed like a RPC call:

```
{
  "method" : "Say",
  "args" : {
    "Peer": 1,
    "Message": "hello"
  }
}
```

4. Any LNDC node can deliver a message to the lit node (message type `0xB0`) that contains:

- PubKey
- JSON
- Signature

5. The lit node verifies if the pubkey is authorized
6. The lit node verifies the signature
7. The lit node carries out the command in the JSON by sending it to its RPC server
8. The lit node returns the response in a message type `0xB1`

I've tested this with the `Say` command so far, which works. Tried with invalid pubkeys and signatures the requests get rejected properly

Next steps:

- [ ] Add IDs to match request/response
- [ ] Allow more finegrained authorization than just "true / false"
- [ ] Add nonces to the requests and disallow double nonces to prevent replays

### CODE REVIEW

**Comment by Varunram** [comment]
retest this please

File: cmd/lit-af/netcmds.go [hunk 1]
```diff
@@ -164,3 +180,50 @@ func (lc *litAfClient) Say(textArgs []string) error {
...
+
+func (lc *litAfClient) RemoteControlAuth(textArgs []string) error {
+	stopEx, err := CheckHelpCommand(rcAuthCommand, textArgs, 2)
+	if err != nil || stopEx {
+		return err
+	}
+	args := new(litrpc.RCAuthArgs)
+	reply := new(litrpc.StatusReply)
+
+	pubKey, err := hex.DecodeString(textArgs[0])
```
Comment (Varunram) [inline cmd/lit-af/netcmds.go]: hey, just trying this out, why do we use a hex encoded argument here? remnants of the old brontide stuff?

File: lit.go [hunk 2]
```diff
@@ -174,6 +178,16 @@ func main() {
 		node.AutoReconnect(conf.AutoListenPort, conf.AutoReconnectInterval)
 	}
 
+	origin := "http://127.0.0.1/"
```
Comment (Varunram) [inline lit.go]: I assume this is for testing direct conns to lit-af?

Comment (Varunram) [inline lit.go]: I assume this is for testing direct conns to lit-af?

File: cmd/lit-af/netcmds.go (hunk 1 above)
Comment (Varunram) [inline cmd/lit-af/netcmds.go]: hey, just trying this out, why do we use a hex encoded argument here? remnants of the old brontide stuff?

File: lndc/conn.go [hunk 3]
```diff
@@ -72,9 +72,24 @@ func parseAdr(netAddress string) (string, string, error) {
 	} else if colonCount >= 5 {
 		conMode = "tcp6"
 		return netAddress, conMode, nil
-	} else {
-		return "", "", fmt.Errorf("Invalid ip")
+	} else if colonCount == 1 {
```
Comment (Varunram) [inline lndc/conn.go]: Nice, I wish we could remove the ugly method of checking ips though, there must be a better way..

**Comment by Varunram** [comment]
retest this please

**Comment by gertjaap** [comment]
I'm gonna redo this.

**Pull request closed without merging**